datafusion/core/tests/dataframe/mod.rs

```diff
@@ -1571,14 +1571,18 @@ async fn with_column_join_same_columns() -> Result<()> {
 
     assert_snapshot!(
         df_with_column.logical_plan(),
-        @r###"
+        @r"
     Projection: t1.c1, t2.c1, Boolean(true) AS new_column
       Limit: skip=0, fetch=1
         Sort: t1.c1 ASC NULLS FIRST
           Inner Join: t1.c1 = t2.c1
-            TableScan: t1
-            TableScan: t2
-    "###
+            SubqueryAlias: t1
```

Comment:
I am concerned about this change -- it seems like a regression to me. 

Maybe it is ok, but I don't understand the difference/ why there are a bunch of projections / SubqueryAlias nows

If these are fixed by physical plan time it is fine, but from here it looks like something is wrong

Reply:
I explain it in the above https://github.com/apache/datafusion/pull/15201#discussion_r1996487546